### DISCUSSION TITLE
PCA9695Expander Example hangs on ESP32

### DISCUSSION THREAD

**Discussion by sjgcornish** [ArminJo/ServoEasing#63] [General]
The example hangs in this for loop after the first iteration of Servo1.startEaseToD(135, 1000); unless I add a small delay i.e. delay(1) or any other code such as print statement in two places as shown in the code below.  Note I am using an ESP32 compiled using DOIT ESP32 DEVKIT V1
    for (int i = 0; i < 2; ++i) {
        Servo1.startEaseToD(135, 1000);
//        Servo1.startEaseToD((544 + (((2400 - 544) / 4) * 3)), 1000); // Alternatively you can specify the target as microsecond value
        // isMoving() calls yield for the ESP8266 boards
        while (Servo1.isMoving()) {
            /*
             * Put your own code here
             */
             delay(1);
            ; // no delays here to avoid break between forth and back movement
        }
        Servo1.startEaseToD(45, 1000);
//        Servo1.startEaseToD((544 + ((2400 - 544) / 4)), 1000); // Alternatively you can specify the target as microsecond value
        while (Servo1.isMoving()) {
            ; // no delays here to avoid break between forth and back movement
        }
        delay(1);
    }

**Comment by ArminJo** [comment]
Thanks for reporting 🥇. It seems to be a ESP32 RTOS bug 😢. I adapted the examples.

**Comment by treppler** [comment]
Hi Armin,

i love your library. I succesfully used it 1 year ago for an animated singing skull for halloween with an Arduino MKR1000.
Now I stuck in the same function. The difference is i'm using a RP2040 clone from Ali.
The programm thinks that the servo is moving. But it doesn't and the LED keeps blinking unregularly (I took it out of the programm and it keeps blinking!?)
<img width="1018" height="427" alt="grafik" src="https://github.com/user-attachments/assets/62e5e47f-4b1e-4559-9e43-040bf1548b3c" />

<img width="511" height="326" alt="grafik" src="https://github.com/user-attachments/assets/8895aec0-34f0-4cf6-aa84-b5dfb06a3c8a" />
It counts 20 times "isch der misch net".
Is there something wrong with the ISR or taskscheduler?

Another thing. Arduino IDE Compiler says:
**WARNUNG: Bibliothek ServoEasing behauptet auf avr, megaavr, sam, samd, esp8266, esp32, stm32, STM32F1, apollo3, mbed, mbed_nano, rp2040 Architektur(en) ausgeführt werden zu können und ist möglicherweise inkompatibel mit Ihrer derzeitigen Platine, welche auf mbed_rp2040 Architektur(en) ausgeführt wird.**
Only with this board selection i can "use" the RP2040 clone.
[https://kunkune.co.uk/shop/compatible-with-arduino/rp2040-board-type-c-4mb-compatible-with-raspberry-pi-pico-micropython/](url)

**Reply by ArminJo** [reply]
1. The link is broken
2. Enable DEBUG or TRACE
3. try https://github.com/earlephilhower/arduino-pico

**Comment by treppler** [comment]
sorry for the blurry.
second try.
<img width="805" height="437" alt="grafik" src="https://github.com/user-attachments/assets/ddebd594-dbe2-4670-8293-ed75851263cc" />

**Comment by treppler** [comment]
Here's the Trace output
<img width="1076" height="292" alt="grafik" src="https://github.com/user-attachments/assets/70457dd6-7b4e-4826-9671-c97e9708b7df" />

**Comment by treppler** [comment]
Thank's for your fast help. Now it works.
<img width="1240" height="1033" alt="grafik" src="https://github.com/user-attachments/assets/61ef81de-7fbb-4ef0-88e8-09445e1bfcfe" />

**Reply by ArminJo** [reply]
Which action did help?

**Reply by treppler** [reply]
Installing (i allready had it):
https://github.com/earlephilhower/arduino-pico
Choosing the right board of this library.
In my case Raspberry Pi Pico.
First i used Generic RP2040 with weiten effect.
